src/poliastro/contrib/extract_czml.py

```diff
@@ -0,0 +1,279 @@
+import json
+
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import CartesianRepresentation
+from astropy.time import Time
+
+from poliastro.contrib.czml_extract_default_params import DEFAULTS
+
+
+class CZMLExtractor:
+    """A class for extracting orbitary data to Cesium"""
+
+    def __init__(self, start_epoch, end_epoch, N):
+        """
+        Orbital constructor
+
+        Parameters
+        ----------
+        start_epoch: ~astropy.time.core.Time
+            Starting epoch
+        end_epoch: ~astropy.time.core.Time
+            Ending epoch
+        N: int
+            Default number of sample points.
+            Unless otherwise specified, the number
+            of sampled data points will be N when calling
+            add_orbit()
+        """
+        self.czml = dict()
+        self.orbits = []
+        self.N = N
+        self.i = 0
+
+        self.start_epoch = CZMLExtractor.format_date(start_epoch)
+        self.end_epoch = CZMLExtractor.format_date(end_epoch)
+
+        self.__init_czml__()
+
+    def parse_dict_tuples(self, path, tups):
+        """
+        Parameters
+        ----------
+        path : list (val)
+            Dictionary path to insert to
+        tups : list (val, val)
+            Tuples to be assigned
+        """
+
+        # We only want to pass a reference czml, then by modifying our reference, we'll be modifying our base dictionary
+        # which allows us to walk through a path of arbitrary length
+
+        curr = self.czml
+        for p in path:
+            curr = curr.setdefault(p, {})
+        for t in tups:
+            curr[t[0]] = t[1]
+
+    def __init_orbit_packet__(self, i):
```

Comment:
```suggestion
    def _init_orbit_packet(self, i):
```

The double underscores are reserved for internal Python methods.